packages/contracts/test/manager/DiamondTest.t.sol

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./DiamondTestSetup.sol";
+
+contract TestDiamond is DiamondSetup {
+
+    function testHasThreeFacets() public {
+        assertEq(facetAddressList.length, 3);
+    }
+
+    function testFacetsHaveCorrectSelectors() public {
+        for (uint i = 0; i < facetAddressList.length; i++) {
+            bytes4[] memory fromLoupeFacet = ILoupe.facetFunctionSelectors(facetAddressList[i]);
+            bytes4[] memory fromGenSelectors =  generateSelectors(facetNames[i]);
+            assertTrue(sameMembers(fromLoupeFacet, fromGenSelectors));
+        }
+    }
+
```

Comment:
we could add more tests for base functions like loupe / diamond cut  ex:
```

    function testShouldCallGetFacets() public prankAs(admin) {
        bool isSupported = DiamondLoupeFacet(address(diamond))
            .supportsInterface(type(IDiamondLoupe).interfaceId);
        assertEq(isSupported, true);
        address[] memory facetAddresses_ = DiamondLoupeFacet(address(diamond))
            .facetAddresses();
        assertEq(facetAddresses_.length, 3);
    }
}
```

Reply:
we would also need test to check that appstorage storage has not been affected by an upgrade 
e.g. 
let say that we upgrade the diamond with an appstorage like this 
```
struct AppStorage {
    address twapOracleAddress;
    address debtCouponAddress;
    address dollarTokenAddress; // uAD
    address couponCalculatorAddress;
    address dollarMintingCalculatorAddress;
    address bondingShareAddress;
    address bondingContractAddress;
    address stableSwapMetaPoolAddress;
    address curve3PoolTokenAddress; // 3CRV
    address treasuryAddress;
    address governanceTokenAddress; // uGOV
    address sushiSwapPoolAddress; // sushi pool uAD-uGOV
    address masterChefAddress;
    address formulasAddress;
    address creditTokenAddress; // uCR
    address newAddressThatWillBeOverwritePreviousStorageSlot; // uCR calculator
    address ucrCalculatorAddress; // uCR calculator
    // couponmanager => excessdollardistributor
    mapping(address => address) _excessDollarDistributors;
}
```
We should be able to run previously working test with the upgraded contract 
In this example the `ucrCalculatorAddress` would be zero instead of ucrCalculator address

Reply:
good catch! @zgorizzo69. Its common experience for storage values to be affected during the upgrade. That would be awesome we can make sure it with the unit test

Reply:
@ryzhak might be able to lend a hand here @0xcodercrane your decision on this.

Reply:
I think its a little bit different from simple unit tests. @sunny0714 's doing good job. I wanna believe sunny will handle conversations as we want.